From ac0829db3fc5d97bbbe5ad0704ead21d4bf42ced Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 16 Nov 2012 12:51:59 +0800
Subject: [PATCH 2/2] ENGR00233894 Available length does't match with the
 actual length in recording

The default bitrate in StageFrightRecoder is 12200, but SoundRecorder use
5900 as the default bitrate, they are not match. So update the bitrate in the
SoundRecorder.
Use 12200 will have better audio quality.

Change-Id: I92c1ae498683c2a73222f0d445f389cdc49c39fd
Signed-off-by: b02247 <b02247@freescale.com>
---
 src/com/android/soundrecorder/Recorder.java      | 2 ++
 src/com/android/soundrecorder/SoundRecorder.java | 5 +++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/com/android/soundrecorder/Recorder.java b/src/com/android/soundrecorder/Recorder.java
index b9654cd..9266f4e 100644
--- a/src/com/android/soundrecorder/Recorder.java
+++ b/src/com/android/soundrecorder/Recorder.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2011 The Android Open Source Project
+ * Copyright (C) 2012 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -164,6 +165,7 @@ public class Recorder implements OnCompletionListener, OnErrorListener {
         mRecorder.setOutputFormat(outputfileformat);
         mRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
         mRecorder.setOutputFile(mSampleFile.getAbsolutePath());
+        mRecorder.setAudioEncodingBitRate(SoundRecorder.BITRATE_AMR);
 
         // Handle IOException
         try {
diff --git a/src/com/android/soundrecorder/SoundRecorder.java b/src/com/android/soundrecorder/SoundRecorder.java
index 2ae2c09..69953aa 100644
--- a/src/com/android/soundrecorder/SoundRecorder.java
+++ b/src/com/android/soundrecorder/SoundRecorder.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2011 The Android Open Source Project
+ * Copyright (C) 2012 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -204,8 +205,8 @@ public class SoundRecorder extends Activity
     static final String AUDIO_ANY = "audio/*";
     static final String ANY_ANY = "*/*";
     
-    static final int BITRATE_AMR =  5900; // bits/sec
-    static final int BITRATE_3GPP = 5900;
+    static final int BITRATE_AMR =  12200; // bits/sec
+    static final int BITRATE_3GPP = 12200;
     
     WakeLock mWakeLock;
     String mRequestedType = AUDIO_ANY;
-- 
1.8.0

